### DISCUSSION TITLE
How to choose the right version of Milvus

### DISCUSSION THREAD

**Discussion by ghost** [milvus-io/milvus#24913] [Q&A and General discussion] [answered]
When connection Milvus container with Python SDK, `<MilvusException: (code=1, message=this version of sdk is incompatible with server, please downgrade your sdk or upgrade your server )>` occurs.

**Comment by xiaofan-luan** [comment, marked as answer]
> When connection Milvus container with Python SDK, `<MilvusException: (code=1, message=this version of sdk is incompatible with server, please downgrade your sdk or upgrade your server )>` occurs.

you can find from https://milvus.io/docs/release_notes.md#Release-Notes

**Reply by ghost** [reply]
My `pymilvus` version is 2.2.9 and Milvus-Standalone is of V2.2.3. Maybe it is a bug that `pymilvus` is not backward-compatible in 2.2.x series.

**Reply by xiaofan-luan** [reply]
I will recommend to use older sdk version with later server version

**Comment by Anaisdg** [comment]
Hello thanks all for your help. 
I'm encountering the same issue, but can't seem to install a previous version of pymilvus successfully either. Do you have any recommendations for the older sdk version I should try?

**Reply by ghost** [reply]
I actually succeeded by installing exact version according to the [release notes](milvus.io/docs/release_notes.md#Release-Notes).
